clippy_dev/src/fmt.rs

```diff
@@ -291,7 +291,12 @@ fn run_rustfmt(context: &FmtContext) -> Result<(), Error> {
             let entry = entry.expect("failed to find tests");
             let path = entry.path();
 
-            if path.extension() != Some("rs".as_ref()) || entry.file_name() == "ice-3891.rs" {
+            if path.extension() != Some("rs".as_ref())
+                || path
+                    .components()
+                    .any(|component| component.as_os_str() == "syntax-error-recovery")
```

Comment:
Rather than checking any component, you can check the one just before the file name:

```suggestion
                || path
                    .components()
                    .rev()
                    .nth(1)
                    .is_some_and(|c| c.as_os_str() == "syntax-error-recovery")
```